sdk/src/main/java/com/horizen/account/api/rpc/types/EthereumBlockView.java

```diff
@@ -57,13 +68,12 @@ public EthereumBlockView(Long blockNumber, String hash, boolean hydratedTx, Acco
         this.timestamp = Numeric.prependHexPrefix(Long.toHexString(block.timestamp()));
         this.baseFeePerGas = Numeric.toHexStringWithPrefix(blockHeader.baseFee());
 
-        var transactions = scala.collection.JavaConverters.seqAsJavaList(block.transactions());
+        var transactions = scala.collection.JavaConverters.seqAsJavaList(AccountBlockUtil.ethereumTransactions(block.transactions()));
 
         if (!hydratedTx) {
-            this.transactions = transactions.stream().map(t -> Numeric.prependHexPrefix((String) t.id())).collect(Collectors.toList());
-        }
-        else {
-            this.transactions = transactions;
+            this.transactions = transactions.stream().map(t -> Numeric.prependHexPrefix(t.id())).collect(Collectors.toList());
+        } else {
+            this.transactions = transactions.stream().map(t -> new EthereumTransactionView(null, t, block.header().baseFee())).collect(Collectors.toList());
```

Comment:
I don't think the constructor of `EthereumTransactionView` currently accepts `null`, it will throw NullPointerException.